src/SpeechModule.ts

```diff
@@ -1,43 +1,169 @@
-
 export type Action = () => void
 
-export type Func<T,TResult> = (item: T) => TResult;
+export type Func<T, TResult> = (item: T) => TResult;
+
+export module Speech {
+    export interface ISpeechRecognizer {
+        locale: string;
+        isStreamingToService: boolean;
+        referenceGrammarId: string; // unique identifier to send to the speech implementation to bias SR to this scenario
 
-export class SpeechSynthesizer {
-    private static instance : SpeechSynthesis.ISpeechSynthesizer = null;
+        onIntermediateResult: Func<string, void>;
```

Comment:
because it will have the intermediate recognition result as an input to this event handler. 
One scenario is when doing recognition, the intermediate result can be shown in the text box so that user can "see" the recognition process.